recipes-bsp/firmware-imx/firmware-nxp-wifi_1.0.bb

```diff
@@ -25,66 +25,8 @@ do_compile() {
-
-    # Install NXP Connectivity PCIE8997 firmware
-    install -m 0644 nxp/FwImage_8997/ed_mac_ctrl_V3_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997/pcie8997_wlan_v4.bin      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997/pcieuart8997_combo_v4.bin ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997/txpwrlimit_cfg_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997/uartuart8997_bt_v4.bin    ${D}${nonarch_base_libdir}/firmware/nxp
-
-    # Install NXP Connectivity SDIO8997 firmware
-    install -m 0644 nxp/FwImage_8997_SD/ed_mac_ctrl_V3_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997_SD/sdio8997_wlan_v4.bin      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997_SD/sdiouart8997_combo_v4.bin ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_8997_SD/txpwrlimit_cfg_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-
-    # Install NXP Connectivity PCIE9098 firmware
-    install -m 0644 nxp/FwImage_9098_PCIE/ed_mac_ctrl_V3_909x.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_9098_PCIE/pcie9098_wlan_v1.bin      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_9098_PCIE/pcieuart9098_combo_v1.bin ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_9098_PCIE/txpwrlimit_cfg_9098.conf  ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_9098_PCIE/uartuart9098_bt_v1.bin    ${D}${nonarch_base_libdir}/firmware/nxp
-
-    # Install NXP Connectivity SDIO9098 firmware
-    install -m 0644 nxp/FwImage_9098_SD/sdio9098_wlan_v1.bin      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_9098_SD/sdiouart9098_combo_v1.bin ${D}${nonarch_base_libdir}/firmware/nxp
-
-    # Install NXP Connectivity IW416 firmware
-    install -m 0644 nxp/FwImage_IW416_SD/sdioiw416_wlan_v0.bin      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_IW416_SD/sdiouartiw416_combo_v0.bin ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_IW416_SD/uartiw416_bt_v0.bin        ${D}${nonarch_base_libdir}/firmware/nxp
-
-    # Install NXP Connectivity IW612 firmware
-    install -m 0644 nxp/FwImage_IW612_SD/sduart_nw61x_v1.bin.se ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_IW612_SD/sd_w61x_v1.bin.se      ${D}${nonarch_base_libdir}/firmware/nxp
-    install -m 0644 nxp/FwImage_IW612_SD/uartspi_n61x_v1.bin.se ${D}${nonarch_base_libdir}/firmware/nxp
-    for f in nxp/FwImage_IW612_SD/IW612_SD_RFTest/*; do
-        install -D -m 0644 $f ${D}${nonarch_base_libdir}/firmware/nxp/IW612_SD_RFTest/$(basename $f)
-    done
+    install -d ${D}${nonarch_base_libdir}/firmware/nxp
+    oe_runmake install INSTALLDIR=${D}${nonarch_base_libdir}/firmware/nxp
```

Comment:
I believe that @junzhuimx did already make a request for this change, but @otavio did not want it. Perhaps it was in email because I do not see a PR.

Reply:
@thochstein, your memory is better than mine. Is it because some firmware exists in `linux-firmware`?

Reply:
I have compared the resultant installation files, and there is 2 differences.

the lines:

```
    for f in cyw-wifi-bt/*_CYW*/brcmfmac4359-pcie*; do
        install -D -m 0644 $f ${D}${nonarch_base_libdir}/firmware/brcm/$(basename $f)
    done

    for f in cyw-wifi-bt/*_CYW*/BCM4349B1*.hcd; do
        install -D -m 0644 $f ${D}${sysconfdir}/firmware/$(basename $f)
    done
```

Should be kept.

And the package `firmware-nxp-wifi-nxpiw612-sdio` has less files. But it look like the files are missing, and not that they are not being installed.
[firmware-wifi-nxp-PR.txt](https://github.com/Freescale/meta-freescale/files/14213615/firmware-wifi-nxp-PR.txt)
[firmware-wifi-nxp-master.txt](https://github.com/Freescale/meta-freescale/files/14213616/firmware-wifi-nxp-master.txt)


Attached are the packages-split for when using master, or using the PR

Reply:
For completeness:

```diff
--- #<buffer firmware-wifi-nxp-master.txt>
+++ #<buffer firmware-wifi-nxp-PR.txt>
@@ -1,15 +1,6 @@
 packages-split/
 ├── firmware-nxp-wifi
 ├── firmware-nxp-wifi-bcm4359-pcie
-│   ├── etc
-│   │   └── firmware
-│   │       └── BCM4349B1_002.002.014.0162.0370.hcd
-│   └── lib
-│       └── firmware
-│           └── brcm
-│               ├── brcmfmac4359-pcie.bin
-│               ├── brcmfmac4359-pcie.clm_blob
-│               └── brcmfmac4359-pcie.txt
 ├── firmware-nxp-wifi-dbg
 ├── firmware-nxp-wifi-dev
 ├── firmware-nxp-wifi-doc
@@ -71,6 +62,7 @@
 │   └── lib
 │       └── firmware
 │           └── nxp
+│               ├── helper_uart_3000000.bin
 │               └── wifi_mod_para.conf
 ├── firmware-nxp-wifi-nxpiw416-sdio
 │   └── lib
@@ -83,14 +75,10 @@
 │   └── lib
 │       └── firmware
 │           └── nxp
-│               ├── IW612_SD_RFTest
-│               │   ├── sduart_nw61x_rftm_v1.bin.se
-│               │   ├── sd_w61x_rftm_v1.bin.se
-│               │   └── uartspi_n61x_v1.bin.se
 │               ├── sduart_nw61x_v1.bin.se
 │               ├── sd_w61x_v1.bin.se
 │               └── uartspi_n61x_v1.bin.se
 ├── firmware-nxp-wifi-src
 └── firmware-nxp-wifi-staticdev
 
-58 directories, 35 files
+52 directories, 29 files

```

Reply:
Oh, sorry, I did confuse multiple changes here. It sounds like you've got it straightened out with adding the BRCM manual install lines. Apparently, they didn't include BRCM install in the new install target.

Reply:
only to make sure I take note:
[firmware-wifi-nxp-PR2.txt](https://github.com/Freescale/meta-freescale/files/14214489/firmware-wifi-nxp-PR2.txt)